From 62c4b5f114850d39efdca3df3b9ee83cc0f424ff Mon Sep 17 00:00:00 2001
From: Jeremy Koritzinsky <jekoritz@microsoft.com>
Date: Tue, 21 Jan 2025 16:16:49 -0800
Subject: [PATCH] Use the documented Artifact extension point to add artifacts

The Artifact extension point handles symbol packages automatically for source-build, and it automatically gets support for new features like Artifact Visibility (required for https://github.com/dotnet/sdk/pull/46063)

Backport: https://github.com/NuGet/NuGet.Client/pull/6233
---
 eng/Publishing.props | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/eng/Publishing.props b/eng/Publishing.props
index 38810c73428..1175200dd59 100644
--- a/eng/Publishing.props
+++ b/eng/Publishing.props
@@ -1,11 +1,8 @@
 <Project>

   <ItemGroup>
-    <ItemsToPushToBlobFeed Include="$(ArtifactsDir)nupkgs/*.nupkg"
-                           IsShipping="true"
-                           UploadPathSegment="nuget-client" />
-    <_SymbolsPackages Include="$(ArtifactsDir)nupkgs/*.symbols.nupkg" />
-    <ItemsToPushToBlobFeed Remove="@(_SymbolsPackages)" Condition="'$(DotNetBuildSourceOnly)' == 'true'" />
+    <Artifact Include="$(ArtifactsDir)nupkgs/*.nupkg"
+              PublishFlatContainer="false" />
   </ItemGroup>

-</Project>
\ No newline at end of file
+</Project>
